case/src/main/kotlin/com/ritense/case_/service/CaseWidgetTabService.kt

```diff
@@ -69,6 +78,34 @@ class CaseWidgetTabService(
         return CaseWidgetTabDto.of(caseWidgetTabRepository.save(caseWidgetTab), caseWidgetMappers)
     }
 
+    @Transactional(readOnly = true)
+    fun getCaseWidgetData(caseDefinitionName: String, tabKey: String, widgetKey: String): Any? {
+        checkCaseTabAccess(caseDefinitionName, tabKey, VIEW)
+
+        val widgetTab = caseWidgetTabRepository.findByIdOrNull(CaseTabId(caseDefinitionName, tabKey)) ?: return null
+        val widgetConfig = widgetTab.widgets.firstOrNull { it.key == widgetKey } ?: return null
+
+        //TODO: Do some authorization check on the widget here?
+
+        return caseWidgetDataProviders.firstNotNullOf { provider ->
+            provider.getData(widgetTab, widgetConfig)
+        }
+    }
+
+    private fun checkCaseTabAccess(caseDefinitionName: String, key: String, action: Action<CaseTab>) {
+        if (!AuthorizationContext.ignoreAuthorization) {
```

Comment:
The ignore authorization is done on another level (by PBAC) already. Why is this check needed here?

Reply:
Performance reasons. We're only using the result of `caseTabRepository.findByIdOrNull` for the PBAC check.